File: Source/C#/DevnologyFitnesseDojo/DevnologyFitnesseDojo/Fitnesse/Amasun/Book/InitializeBookInventory.cs
using DevnologyFitnesseDojo.Domain;
using DevnologyFitnesseDojo.Domain.Books;

namespace DevnologyFitnesseDojo.Fitnesse.Amasun.Book
{
    public class InitializeBookInventory
    {
        private int amount;
        private string author;
        private string isbn;
        private decimal price;
        private string title;

        public void Execute()
        {
            var book = new Domain.Books.Book { Author = author, Title = title, Isbn = isbn, Price = price };

            Inventory.AddBook(book, amount);
        }

        public void SetAuthor(string author)
        {
            this.author = author;
        }

        public void SetTitle(string title)
        {
            this.title = title;
        }

        public void SetIsbn(string isbn)
        {
            this.isbn = isbn;
        }

        public void SetPrice(decimal price)
        {
            this.price = price;
        }

        public void SetAmount(int amount)
        {
            this.amount = amount;
        }
    }
}

File: Source/Java/nl/devnology/fitnesse/amasun/book/InitializeBookInventory.java
package nl.devnology.fitnesse.amasun.book;

import nl.devnology.domain.book.Book;
import nl.devnology.domain.book.Inventory;

/**
 * @author Erik Pragt
 */
public class InitializeBookInventory {
    private String author;
    private String title;
    private String isbn;
    private Integer price;
    private Integer amount;

    public void execute() {
        Book book = new Book(author, title, isbn, price);

        Inventory inventory = Inventory.get();

        inventory.addBook(book, amount);
    }

    public void setAuthor(String author) {
        this.author = author;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public void setIsbn(String isbn) {
        this.isbn = isbn;
    }

    public void setPrice(Integer price) {
        this.price = price;
    }

    public void setAmount(int amount) {
        this.amount = amount;
    }
}
